test/cls_test.py

```diff
@@ -878,6 +878,7 @@ class UsingAnnotationParameters:
     a: int = modal.parameter()
     b: str = modal.parameter(default="hello")
     c: float = modal.parameter(init=False)
+    d: Any = modal.parameter(default={"foo": "bar"})
```

Comment:
Thinking some more about this I think it might be better to **not** type these as `Any`, but instead `typing.Annotated[{ActualType}, modal.PickleSerialization]` or similar.

Otherwise users can't both get good typing support in their editor and support for non-supported types.

Reply:
I.e. then users can do:

```
arr: Annotated[pandas.DataFrame, modal.PickleSerialization] = modal.parameter()
```

Reply:
Added!